reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -351,43 +356,42 @@ static boolean hasOutboundTerminated(long state) {
 	static abstract class AbstractSwitchOnFirstMain<T, R> extends Flux<T>
 			implements InnerOperator<T, R> {
 
-		final ControlSubscriber<? super R>                                     outer;
-		final BiFunction<Signal<? extends T>, Flux<T>, Publisher<? extends R>>
-		                                                                       transformer;
+		final ControlSubscriber<? super R>                                     outboundSubscriber;
+		final BiFunction<Signal<? extends T>, Flux<T>, Publisher<? extends R>> transformer;
 
 		Subscription s;
 
 		boolean isInboundRequestedOnce;
+		boolean isFirstOnNextValueReceivedOnce;
```

Comment:
leftover from tentative change

Reply:
it could actually be a follow-up change, but I think for now it is fine to still store the onNext value in `this.firstValue`, provided we null it out in all places where we discard it.

Reply:
done